windows/code/RNDeviceInfoCPP.h

```diff
@@ -75,6 +153,79 @@ namespace winrt::RNDeviceInfoCPP
         });
     }
 
+    REACT_SYNC_METHOD(isKeyboardConnectedSync);
+    bool isKeyboardConnectedSync() noexcept
+    {
+      auto keyboardCapabilities = winrt::Windows::Devices::Input::KeyboardCapabilities();
+      return keyboardCapabilities.KeyboardPresent();
+    }
+
+    REACT_METHOD(isKeyboardConnected);
+    void isKeyboardConnected(ReactPromise<bool> promise) noexcept
+    {
+      promise.Resolve(isKeyboardConnectedSync());
+    }
+
+    REACT_SYNC_METHOD(isMouseConnectedSync);
+    bool isMouseConnectedSync() noexcept
+    {
+      auto mouseCapabilities = winrt::Windows::Devices::Input::MouseCapabilities();
+      return mouseCapabilities.MousePresent();
+    }
+
+    REACT_METHOD(isMouseConnected);
+    void isMouseConnected(ReactPromise<bool> promise) noexcept
+    {
+      promise.Resolve(isMouseConnectedSync());
+    }
+
+    // Helper function, should eventually be removed/replaced when it's exposed through RNW's public API.
+    bool IsXamlIsland() {
+      AppPolicyWindowingModel e;
+      // This value comes from an inline function GetCurrentThreadEffectiveToken() from processthreadsapi.h
+      auto token = (HANDLE)(LONG_PTR) -6;
+      auto windowingModel = CALL_INDIRECT(L"Api-ms-win-appmodel-runtime-l1-1-2.dll", AppPolicyGetWindowingModel, token, &e);
+      if (FAILED(windowingModel) || e == AppPolicyWindowingModel_ClassicDesktop) {
```

Comment:
can't do FAILED here since AppPolicyGetWindowingModel doesn't return an HRESULT sadly
Check for != ERROR_SUCCESS instead

Reply:
Updated!